contracts/interchain-token-service/src/contract.rs

```diff
@@ -66,49 +75,53 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
             .expect("interchain token wasm hash not found")
     }
 
-    fn its_hub_routing_identifier(env: &Env) -> String {
-        String::from_str(env, ITS_HUB_ROUTING_IDENTIFIER)
+    fn its_hub_address(env: &Env) -> String {
+        env.storage()
+            .instance()
+            .get(&DataKey::ItsHubAddress)
+            .expect("its hub address not found")
     }
 
     fn its_hub_chain_name(env: &Env) -> String {
         String::from_str(env, ITS_HUB_CHAIN_NAME)
     }
 
-    fn trusted_address(env: &Env, chain: String) -> Option<String> {
+    fn is_trusted_chain(env: &Env, chain: String) -> bool {
         env.storage()
             .persistent()
-            .get(&DataKey::TrustedAddress(chain))
+            .has(&DataKey::TrustedChain(chain))
     }
 
-    fn set_trusted_address(env: &Env, chain: String, address: String) -> Result<(), ContractError> {
+    fn set_trusted_chain(env: &Env, chain: String) -> Result<(), ContractError> {
         Self::owner(env).require_auth();
 
-        let key = DataKey::TrustedAddress(chain.clone());
+        let key = DataKey::TrustedChain(chain.clone());
 
         ensure!(
             !env.storage().persistent().has(&key),
-            ContractError::TrustedAddressAlreadySet
+            ContractError::TrustedChainAlreadySet
         );
 
-        env.storage().persistent().set(&key, &address);
+        env.storage().persistent().set(&key, &true);
```

Comment:
```suggestion
        env.storage().persistent().set(&key, &());
```
Store unit type instead